src/control/controlcompressingproxy.cpp

```diff
@@ -0,0 +1,33 @@
+#include "control/controlcompressingproxy.h"
+
+#include "moc_controlcompressingproxy.cpp"
+
+// Event queue compressing proxy
+CompressingProxy::CompressingProxy(QObject* parent)
+        : QObject(parent) {
+}
+
+// This function is called recursive by QCoreApplication::sendPostedEvents, until no more events are in the queue.
+// When the last event is found, QCoreApplication::sendPostedEvents returns for the isLatestEventInQueue() instance of this event,
+// and m_recursiveSearchForLastEventOngoing is set to false, while returning true itself.
+// All previous started instances of isLatestEventInQueue() will return false in consequence,
+// because the return value depends on the member variable and not on the stack of the instance.
+bool CompressingProxy::isLatestEventInQueue() {
```

Comment:
*is* implies *idempotent* for me.

How about `processQueuedEvents()` that returns either *last event* or *outdated event* encoded in some `enum class`.

Reply:
Done!